src/Http/Logger/LoggerServiceProvider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OWC\Zaaksysteem\Http\Logger;
+
+use OWC\Zaaksysteem\Foundation\ServiceProvider;
+use OWC\Zaaksysteem\GravityForms\GravityFormsSettings;
+
+class LoggerServiceProvider extends ServiceProvider
+{
+    protected string $settingsPrefix = OWC_GZ_PLUGIN_SLUG;
+
+    public function boot(): void
+    {
+        $this->setLoggingLevel();
+
+        if ($this->isEnabled()) {
+            add_action('http_api_debug', [$this, 'logHttpMessage'], 10, 5);    
+        }
+
+        add_filter('owc_gravityforms_zaaksysteem_gf_settings', [$this, 'addLoggingOptions']);
+    }
+
+    public function isEnabled(): bool
+    {
+        $container = $this->plugin->getContainer();
+
+        return (bool) $container->get('message.logger.active');
+    }
+
+    public function logHttpMessage($response, $context, $class, $arguments, $url)
+    {
+        if (! ($arguments['headers']['_owc_request_logging'] ?? false)) {
+            return;
+        }
+
+        $this->plugin->getContainer()->get('message.logger')->debug($url, compact('arguments', 'response'));
+    }
+
+    public function addLoggingOptions(array $fields): array
+    {
+        // Insert logging fields after the 'Algemene instellingen' fields.
+        $columns = array_column($fields, 'title');
+        $index = (array_flip($columns)['Algemene instellingen'] ?? 0) + 1;
+        
+        return array_merge(
+            array_slice($fields, 0, $index), 
+            [$this->getFieldSettings()],
+            array_slice($fields, $index, count($fields))
+        );
+    }
+
+    protected function getFieldSettings(): array
+    {
+        return [
+            'title'  => esc_html__('Enable message logging', 'owc-gravityforms-zaaksysteem'),
```

Comment:
Gezien in deze plugin (volgens mij) alles standaard in NL staat is het misschien handig om dat hier ook te doen? Ik zou white box etc niet vertalen... :p

Reply:
"Witte doos logboeken" is toch 'n prachtige vertaling 😄 Maar eens, heb er maar laag/gemiddeld/hoog van gemaakt.